### ISSUE TITLE
[Terminal Canary] The language setting doesn't match with the OS

### ISSUE THREAD

**Issue by AvogatoWizardWhisker** [microsoft/terminal#17480]
### Windows Terminal version

1.22.1781.0

### Windows build number

10.0.26241.5000

### Steps to reproduce

1. I changed the language setting from English to French.
2. The entire OS changed to French after I signed out.
3. I opened Terminal Canary and it's still in English.

![image](https://github.com/microsoft/terminal/assets/89489853/b248d5be-4b3d-4e16-a009-4fa85a0055ad)

### Expected Behavior

I'm expecting the language in Terminal to match with the OS since it should follow the system default

### Actual Behavior

Even tho Terminal's setting language uses the "system default", it remains the same previous language.

**Comment by DHowett** [comment]
Is this still happening/? We haven't had any other reports on Canary of this happening, and that build is a couple months out of date.

**Comment by AvogatoWizardWhisker** [comment]
I'll try to retry again, and i'll let you know if the problem presists.

**Comment by AvogatoWizardWhisker** [comment]
I apologize for the late response. I tried that again with Terminal Canary version: 1.23.2961.0. The issue still persists.

![Image](https://github.com/user-attachments/assets/1d233edf-a3f6-4ed1-9065-bbc4546cca83)
 
Do you think it has to do with administrative settings?

![Image](https://github.com/user-attachments/assets/b449092f-55bf-4ca8-a086-ef95b4117305)

**Comment by lhecker** [comment]
I don't believe so. The first item in each of these ("Display Language" in English) should be the one what the app platform should use for determining the language for WinUI. I think the language settings in the registry for Windows Terminal may have gone out of sync with the rest of the OS for you.
@DHowett for instance encountered a bug just yesterday where he changed the "accent color" (the window border frame color) and Windows Terminal did not adopt it. He checked the package registry and it still contained the old value. I think these may just be various bugs with the packaged app / app platform implementation on Windows. 🤔

As you probably already know or noticed, you can change the "Language" setting in the left half of your screenshot (roughly in the center) to French and it should work around the issue. I'm not sure if we can fix this on our side since we rely on WinUI which in turn relies on the [`ApplicationLanguages.Languages`](https://learn.microsoft.com/en-us/uwp/api/windows.globalization.applicationlanguages.languages) API.

**Comment by carlos-zamora** [comment]
Sorry we didn't ask this earlier, but have you tried uninstalling and reinstalling Windows Terminal /? Make sure you save your settings before you uninstall it though! Reinstalling the app should reset the registry and fix the issue.

**Comment by AvogatoWizardWhisker** [comment]
> Sorry we didn't ask this earlier, but have you tried uninstalling and reinstalling Windows Terminal /? Make sure you save your settings before you uninstall it though! Reinstalling the app should reset the registry and fix the issue.

I apologise for the delay. I did try uninstalling and reinstalling Terminal Canary while Windows is in French. Even though Terminal uses the default settings for languages, it still shows in English rather than French. 

![Image](https://github.com/user-attachments/assets/36bde044-9ae6-41a6-8aed-b06389fd7cf8)

Is there some changes I need to make in administrative settings?

![Image](https://github.com/user-attachments/assets/d7665f16-6a19-473a-9c07-2321d5b430aa)
